timesketch/lib/utils.py

```diff
@@ -36,87 +36,82 @@ def random_color():
-    mandatory_fields = [u'Alert', u'Tag', u'Timestamp', u'Field', u'Summary']
+    Returns:
+        Generator of event rows
 
+    Raises:
+        RuntimeError is CSV format is unknown
+    """
     with open(path, 'rb') as fh:
-        csv.register_dialect('myDialect',
-                             delimiter=',',
-                             quoting=csv.QUOTE_ALL,
-                             skipinitialspace=True)
-        reader = csv.DictReader(fh, delimiter=',', dialect='myDialect')
-
+        reader = csv.DictReader(fh, delimiter=delimiter.decode('string_escape'))
         csv_header = reader.fieldnames
-        missing_fields = []
-        # Validate the CSV header
-        for field in mandatory_fields:
-            if field not in csv_header:
-                missing_fields.append(field)
-        if missing_fields:
-            raise RuntimeError(
-                u'Missing fields in CSV header: {0:s}'.format(missing_fields))
-        for row in reader:
-
-            dt = parser.parse(row['Timestamp'])
-            timestamp = int(time.mktime(dt.timetuple())) * 1000
-            dt_iso_format = dt.isoformat()
-            timestamp_desc = row['Field']
-
-            summary = row['Summary']
-            alert = row['Alert']
-            tag = row['Tag']
-
-            row_to_yield = {}
-            row_to_yield["message"] = summary
-            row_to_yield["timestamp"] = timestamp
-            row_to_yield["datetime"] = dt_iso_format
-            row_to_yield["timestamp_desc"] = timestamp_desc
-            row_to_yield["alert"] = alert #extra field
-            tags = [tag]
-            row_to_yield["tag"] = tags # extra field
-
-            yield row_to_yield
+        csv_dialect = get_csv_dialect(csv_header)
+
+        if u'redline' in csv_dialect:
+            for row in reader:
+                parsed_datetime = parser.parse(row[u'Timestamp'])
+                timestamp = int(
+                    time.mktime(parsed_datetime.timetuple())) * 1000000
+                parsed_datetime_iso_format = parsed_datetime.isoformat()
+                row = dict(
+                    message=row[u'Summary'],
+                    timestamp=timestamp,
+                    datetime=parsed_datetime_iso_format,
+                    timestamp_desc=row[u'Field'],
+                    alert=row[u'Alert'],
+                    tag=[row[u'Tag']]
+                )
+                yield row
+
+        elif u'timesketch' in csv_dialect:
+            for row in reader:
+                if u'timestamp' not in csv_header and u'datetime' in csv_header:
+                    try:
+                        parsed_datetime = parser.parse(row[u'datetime'])
+                        row[u'timestamp'] = int(
+                            time.mktime(parsed_datetime.timetuple())) * 1000000
+                    except ValueError:
+                        continue
+                yield row
+        else:
+            raise RuntimeError(u'Unknown CSV format')
```

Comment:
You can turn this into a guard statement before the other checks

```
if csv_dialect is None:
    raise RuntimeError()

if csv_dialect == 'timesketch':
    ...

if csv_dialect == 'redline':
    ...
```

Reply:
I like the proposal